pkg/cloudprovider/aws/instance.go

```diff
@@ -103,30 +104,19 @@ func (p *InstanceProvider) launchInstance(ctx context.Context,
 	launchTemplateName string,
 	instanceTypeOptions []cloudprovider.InstanceType,
 	subnets []*ec2.Subnet,
-	capacityType string) (*string, error) {
-	// 1. Construct override options.
-	var overrides []*ec2.FleetLaunchTemplateOverridesRequest
-	for i, instanceType := range instanceTypeOptions {
-		for _, zone := range instanceType.Zones() {
-			for _, subnet := range subnets {
-				if aws.StringValue(subnet.AvailabilityZone) == zone {
-					override := &ec2.FleetLaunchTemplateOverridesRequest{
-						InstanceType: aws.String(instanceType.Name()),
-						SubnetId:     subnet.SubnetId,
-					}
-					// Add a priority for spot requests since we are using the capacity-optimized-prioritized spot allocation strategy
-					// to reduce the likelihood of getting an excessively large instance type.
-					// instanceTypeOptions are sorted by vcpus and memory so this prioritizes smaller instance types.
-					if capacityType == v1alpha1.CapacityTypeSpot {
-						override.Priority = aws.Float64(float64(i))
-					}
-					overrides = append(overrides, override)
-					// FleetAPI cannot span subnets from the same AZ, so break after the first one.
-					break
-				}
-			}
-		}
+	capacityTypes []string) (*string, error) {
+
+	// If unconstrained, default to spot to save on cost, otherwise use on
+	// demand. Constraint solving logic will guarantee that capacityTypes are
+	// either unconstrained (nil), or a valid WellKnownLabel. Provisioner
+	// defaulting logic will currently default to [on-demand] if unspecifed.
+	capacityType := v1alpha1.CapacityTypeSpot
+	if capacityTypes != nil && !functional.ContainsString(capacityTypes, v1alpha1.CapacityTypeSpot) {
```

Comment:
interesting, I guess the ICE Cache could key on capacity-type too in order to support a native fallback to OD if we can't get spot capacity?

Reply:
Yeah totally.